App/dojo.py

```diff
@@ -8,163 +8,325 @@
+        time.sleep(1) 
+        rooms = self.all_rooms
+        if not rooms:
+            click.secho('No rooms added yet!.', fg='red', bold=True)
+            return 'No rooms exist at the moment.'
+        for room in rooms:
+            # checks for rooom name that is equal to the name argument given
+            if room_name == room.name:
+                print(" ")
+                click.secho('=' * 30, fg='yellow')
+                click.secho('%s ' %(room_name.capitalize()), fg='white', bold=True)
+                click.secho('=' * 30, fg='yellow')
+                if room.members:
+                    # go through each and every member in a room
+                    for member in room.members:
+                        click.secho(member.name, fg='cyan')
+                else:
+                    # if a created room is empty with no members
+                    click.secho('No members found :(.', fg='cyan', bold=True)
+                click.secho('=' * 30, fg='yellow')
+                print ("    ")
+                return False
+        else:
+            # if no room is created 
+            click.secho('Room %s seems not to exist would you like to add it?'\
+            % (room_name.capitalize()), fg='red', bold=True)
+            print ("    ")
 
     def print_allocations(self, filename):
-        pass
+        '''
+        This function prints a list of members(allocated) of a
+        specific room to the screen and to a file if opted
+        '''
+        # checks for rooms not added or created
+        rooms = self.all_rooms
+        if not rooms:
+            click.secho('No rooms found in the Dojo', bold=True, fg='red')
+            return
+        output = ''
+        for room in rooms:
+            output += '==' * 15
+            output += '\n'
+            output += room.name.upper()
+            output += '\n'
+            output += '==' * 15
+            output += '\n'
+            # go through all members added
+            if room.members:
+                for member in room.members:
+                    output += member.name
+                    output += '\n'
+            else:
+                # prints out a created room with no members
+                output += 'There are no people  yet.'
+                output += '\n'
+        # This prints a list of all allocations onto the screen without the
+        # optional --o
+        if filename == None:
+            click.secho(output, fg='blue')
+
+        else:
+            # prints the list of allocations to a specified file(.txt)
+            click.secho('PLEASE WAIT...', bold=True, fg='magenta')
+            time.sleep(1)
+            print(' ')
+            text = ''
+            for room in rooms:
+                text += room.name.upper() + "\n"
+                text += "-" * 50 + "\n"
+                if room.members:
+                    text += " , " .join(member.name for member in room.members) + '\n'
```

Comment:
@Gchesta for line 265 that's the output format  while 276 line prints out the format to `txt` file